src/main/java/com/WebController/EventController.java

```diff
@@ -63,4 +81,53 @@ public String receivePatient (@RequestBody List<Event> events){
         return list;
     }
 
+    @PostMapping(value = "/userregister", consumes = "application/json")
+    @ResponseBody
+    public String userRegistration(@RequestBody User user){
+        if (this.userRepository.findUserByEmailAddressIgnoreCase(user.getEmailAddress())!=null){
+            return "user already exists";
+        }
+        userRepository.save(user);
+        ConfirmationToken token = new ConfirmationToken(user);
+
+        SimpleMailMessage message = new SimpleMailMessage();
```

Comment:
Avoid hard-coding the content of the email here. Maybe retrieve it from some configuration file. also, avoid using multiple setters for this, and just get the object set up in the constructor.

Reply:
The smell here is that this Controller is getting too "fat". Some of this is Model logic, not Controller logic.